File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_ios/ios/Classes/PigeonBinaryMessengerWrapper.swift
import Foundation
import Flutter

class PigeonBinaryMessengerWrapper: NSObject, FlutterBinaryMessenger {

    let channelSuffix: String
    let messenger: FlutterBinaryMessenger

    init(with messenger: FlutterBinaryMessenger, channelSuffix: String) {
        self.messenger = messenger
        self.channelSuffix = channelSuffix
    }

    func send(onChannel channel: String, message: Data?, binaryReply callback: FlutterBinaryReply? = nil) {
        messenger.send(onChannel: "\(channel)/\(channelSuffix)", message: message, binaryReply: callback)
    }

    func send(onChannel channel: String, message: Data?) {
        messenger.send(onChannel: "\(channel)/\(channelSuffix)", message: message)
    }

    func setMessageHandlerOnChannel(_ channel: String, binaryMessageHandler handler: FlutterBinaryMessageHandler? = nil) -> FlutterBinaryMessengerConnection {
        messenger.setMessageHandlerOnChannel("\(channel)/\(channelSuffix)", binaryMessageHandler: handler)
    }

    func cleanUpConnection(_ connection: FlutterBinaryMessengerConnection) {
        messenger.cleanUpConnection(connection)
    }
}

File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_android/android/src/main/kotlin/com/theoplayer/flutter/PigeonBinaryMessengerWrapper.kt
package com.theoplayer.flutter

import io.flutter.plugin.common.BinaryMessenger
import java.nio.ByteBuffer

// BinaryMessenger that proxies all calls to another BinaryMessenger,
// but with a suffix appended to the channel name.
// Different player instances use different suffixes to avoid channel name conflicts.

class PigeonBinaryMessengerWrapper(
  private val messenger: BinaryMessenger,
  private val channelSuffix: String,
) : BinaryMessenger {

  override fun send(channel: String, message: ByteBuffer?) {
    messenger.send("$channel/$channelSuffix", message)
  }

  override fun send(channel: String, message: ByteBuffer?, callback: BinaryMessenger.BinaryReply?) {
    messenger.send("$channel/$channelSuffix", message, callback)
  }

  override fun setMessageHandler(channel: String, handler: BinaryMessenger.BinaryMessageHandler?) {
    messenger.setMessageHandler("$channel/$channelSuffix", handler)
  }
}